Tests/Services/NfieldTemplatesServiceTests.cs

```diff
@@ -0,0 +1,108 @@
+﻿//    This file is part of Nfield.SDK.
+//
+//    Nfield.SDK is free software: you can redistribute it and/or modify
+//    it under the terms of the GNU Lesser General Public License as published by
+//    the Free Software Foundation, either version 3 of the License, or
+//    (at your option) any later version.
+//
+//    Nfield.SDK is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//    GNU Lesser General Public License for more details.
+//
+//    You should have received a copy of the GNU Lesser General Public License
+//    along with Nfield.SDK.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using Nfield.Infrastructure;
+using Nfield.Services.Implementation;
+using Xunit;
+using Nfield.Models;
+using Moq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Net;
+using Newtonsoft.Json;
+using System.Linq;
+
+namespace Nfield.Services
+{
+    /// <summary>
+    /// Tests for <see cref="NfieldTemplatesService"/>
+    /// </summary>
+    public class NfieldTemplatesServiceTests : NfieldServiceTestsBase
+    {
+        #region QueryAsync
+
+        [Fact]
+        public void TestQueryAsync_ServerReturnsQuery_ReturnsListWithTemplates()
+        {
+            const HttpStatusCode httpStatusCode = HttpStatusCode.NotFound;
+            var expectedTemplates = new Template[]
+            {
+                new Template
+                {
+                    Id = "TemplateId1",
+                    Name = "TemplateName1",
+                    Themes = new Theme[0]
+                },
+                new Template
+                {
+                    Id = "TemplateId2",
+                    Name = "TemplateName2",
+                    Themes = new []
+                    {
+                        new Theme
+                        {
+                            Id = "ThemeId1",
+                            Name = "ThemeName1"
+                        },
+                        new Theme
+                        {
+                            Id = "ThemeId2",
+                            Name = "ThemeName2"
+                        }
+                    }
+                }
+            };
+            var mockedNfieldConnection = new Mock<INfieldConnectionClient>();
+            var mockedHttpClient = new Mock<INfieldHttpClient>();
+            mockedHttpClient
+                .Setup(client => client.GetAsync(new Uri(ServiceAddress, "Templates/")))
+                .Returns(
+                    Task.Factory.StartNew(
+                        () =>
+                            new HttpResponseMessage(httpStatusCode)
```

Comment:
a bit weird that `httpStatusCode` is `HttpStatusCode.NotFound` with some content, and that `QueryAsync` returns valid objects, but I see the SDK pretty much always ignores the the response code.
Maybe use OK instead for the test